### DISCUSSION TITLE
Unusual Results Using MUSCL with SST-V2003m

### DISCUSSION THREAD

**Discussion by j-signorelli** [su2code/SU2#2089] [Q&A]
**Describe the bug**
I noticed I was converging to quite different solutions when MUSCL is on v. off for a much larger case I am doing using SU2 with SST-2003m, so I created a smaller 2D case to test out why this may be occurring. Below is a screenshot of one of the meshes, where the pinkish line indicates a value for an interface that I plot heat fluxes from. I have generated several meshes with increasing level of refinement from A to D.

![image](https://github.com/su2code/SU2/assets/116454159/4d04287e-b111-45b5-96dd-6b9cc5f18217)

It appears that there may be either a misunderstanding on my part with the implementation of SST-V2003m w/ MUSCL, or that a bug exists in SU2 for it. I got very unusual and different results when using MUSCL vs. not. To further look into this, I also completed additional simulations using `KIND_TURB_MODEL=SA`, which got me much more reasonable-looking results. Please see the plots below:

![image](https://github.com/su2code/SU2/assets/116454159/e5d80bbd-9bc2-4f3c-a6e2-bb5dfc43336b)
![image](https://github.com/su2code/SU2/assets/116454159/3804bdf1-7f99-4584-9683-cfbfd7dfa916)

Lastly, here is the config file:
[config.txt](https://github.com/su2code/SU2/files/12155765/M6_LowRe.txt)

Note that the "MUSCL=ON" in the plots indicate the solution for both `MUSCL_FLOW=YES` and `MUSCL_TURB=YES`. That is the only difference between config files. Also, I tried using `MARKER_RIEMANN`, `MARKER_OUTLET`, and `MARKER_SUPERSONIC_OUTLET` on the outlets with no success on fixing this discrepancy, so I believe it is not related to the boundary conditions.

**Bug report checklist**
*Please make sure that you have followed the checklist below, many common problems can be solved by:*

- [X] Consulting the build instructions (https://su2code.github.io/docs_v7/Build-SU2-Linux-MacOS/).
- [x] Looking for similar problems on GitHub or CFD-Online (https://www.cfd-online.com/Forums/su2/).
- [X] Updating to the newest version of SU2 (either develop, master, or the pre-built executables https://su2code.github.io/download.html).

**Desktop (please complete the following information):**
 - OS: [e.g., Linux (CentOS 7)]
 - C++ compiler and version: GCC 9.1.0
 - MPI implementation and version: Intel MPI 19.0.9
 - SU2 Version: v7.5.1 but with following additional commits cherry-picked:
    - 8ce4c3fd5e030949d037e073e1b4418e51c87540
    - 07645b1b70d1dbfcf3d96acfeb10c560532ad60f
    - 54af111101f4aab98cc4589e2242d9cd06730618

**Comment by pcarruscag** [comment]
And when you say MUSCL=OFF is it also for the flow? If so I would not be surprised that a first-order simulation gives different results.

**Comment by j-signorelli** [comment]
Hi @pcarruscag , that is correct. If this is no surprise, then this is no bug. But it seems unusual to me to have this stark of a difference. I rapidly got mesh convergence for SA using MUSCL as shown, but not even close for SST even on additional finer meshes I generated. I also noticed this occurring with pressure:
![image](https://github.com/su2code/SU2/assets/116454159/bfc2d05c-4aab-405e-8e2a-58055600a40c)

**Reply by pcarruscag** [reply]
I would look at whether the mesh you are using has enough resolution in the right places, k-w models have much sharper gradients near the walls than SA.
In theory MUSCL-off will eventually converge to the same answer, it will just take a lof of refinement.

**Reply by sun5k** [reply]
hi @j-signorelli 

In your configuration, you used AUSMPLUSUP2 for convective flux. Did you have the same result as MUSCL on/off, when you use SLAU2 or L2ROE scheme?

**Reply by j-signorelli** [reply]
I ran with what I believe is just an over-excessively fine mesh (labeled Mesh E), and I am still running into this problem. **Note that y+ at the surface varies from Y_Plus=2.9e-4 to a maximum of Y_Plus=0.071.** I also just included an image of the leading edge shock just to help with visualizing the level of refinement I am talking about here. My only concern about mesh quality is that the max aspect ratio is 1000, but skewness and area ratios all are great except for that, and I have an *even finer* mesh running with max AR being ~150.

![image](https://github.com/su2code/SU2/assets/116454159/9c240280-a972-4ab7-bbd9-0c3d522b1220)
![E_MUSCL_LE](https://github.com/su2code/SU2/assets/116454159/13a41249-33dc-4f05-a9e2-a88ae99dbc17)

If any additional thoughts exist for this I would be very interested to hear. Just based on how much the curves are varying for the additional level of refinement for each mesh it looks like it's not even close to convergence.

**Comment by pcarruscag** [comment]
I would also think that would be enough, what kind of convergence do you get for those cases?

**Reply by j-signorelli** [reply]
Here is are residuals for mesh C:

MUSCL_FLOW=OFF, MUSCL_TURB=OFF:

        rms[Rho]|   rms[RhoU]|   rms[RhoV]|   rms[RhoE]|      rms[k]|      rms[w]|
        -12.996597|  -12.094507|  -13.383447|  -11.568134|  -13.028047|   -5.787231|

MUSCL_FLOW=ON, MUSCL_TURB=ON:

        rms[Rho]|   rms[RhoU]|   rms[RhoV]|   rms[RhoE]|      rms[k]|      rms[w]|
        -10.898226|   -9.928328|   -9.983529|   -9.163417|   -9.130655|   -2.151257|

Note that I ran it until the interface heat flux stopped changing to a small precision using Cauchy series convergence criteria. The CFL numbers were also at 10000 for the MUSCL=ON case above upon completion (small CFL did not cause Cauchy series convergence early). I was not able to drive the omega residual smaller. Same thing appears to have occurred for finer meshes.

**Reply by j-signorelli** [reply]
Also the plot below is not for this test case but is for the larger case I mentioned previously, which I think also may be indicative of a bug. It almost appears that 1st-order is converging appropriately for heat flux, but somehow w/ MUSCL is not. Again, not sure if this is just a misunderstanding of how SST operates on my end, but it does not appear correct.
![image](https://github.com/su2code/SU2/assets/116454159/5ff8a212-bed2-4471-bd23-ce1d107cc7a3)

**Reply by pcarruscag** [reply]
I see, can you try with MUSCL_FLOW=ON, MUSCL_TURB=OFF?

**Reply by j-signorelli** [reply]
Thanks for the quick response, I tried this and it yielded very similar curves as to MUSCL on for all:
![image](https://github.com/su2code/SU2/assets/116454159/4a0a7dc8-7192-4d38-bb58-6ebd06fd08f4)

**Reply by pcarruscag** [reply]
Just to see if that drops the w residual a bit more and if that makes a difference in the results

**Reply by pcarruscag** [reply]
But is the w residual lower with MUSCL_TURB off? or still the same?
Also what is the starting residual for all variables?

**Reply by j-signorelli** [reply]
Re-ran mesh A with MUSCL_FLOW=ON, MUSCL_TURB=OFF w/ MUSCL=ON for both solution as IC:

![image](https://github.com/su2code/SU2/assets/116454159/b7e36711-c17c-4a1e-be42-b3e3f0043054)

Initial:

          rms[Rho]|   rms[RhoU]|   rms[RhoV]|   rms[RhoE]|      rms[k]|      rms[w]|
    0|  -11.254863|  -10.253153|  -11.670878|   -9.734594|   -5.340565|   -0.257061|
    1|   -8.789845|   -7.320651|   -7.378766|   -6.471696|   -5.477477|   -0.397589|

Final:

              rms[Rho]|   rms[RhoU]|   rms[RhoV]|   rms[RhoE]|      rms[k]|      rms[w]|
     2811|  -13.041561|  -12.031917|  -13.562713|  -11.474046|  -12.924348|   -6.285397|
